controllers/userController.js

```diff
@@ -40,6 +46,51 @@ const controller = {
       });
     }
   },
+  editProfile: async (req, res) => {
+    const { name, tagline, skills, interests, socmed } = req.body;
+    try {
+      await validator.profile.validateAsync({
+        name,
+        tagline,
+        skills,
+        interests,
+        socmed,
+      });
+    } catch (error) {
+      console.log(error.message);
+      return res.status(400).json({
+        error: 'Invalid input',
+      });
+    }
+    try {
+      const profileOwner = await UserModel.findOne({ username: req.authUser.username });
+      const user = await UserModel.findOne({ username: req.params.username });
+
+      // authorisation check: whether user is the project owner
+      if (user?._id.toString() !== profileOwner._id.toString()) {
+        return res.status(401).json({
+          error: 'User is not authorised to change this project',
+        });
+      }
+
+      const skillsArr = skills
+        .split(',')
+        .map((item) => item.trim())
+        .filter((item) => validSkills.includes(item));
+      const interestsArr = interests.split(',').map((item) => item.trim());
+      const socmedArr = socmed.split(',').map((item) => item.trim());
```

Comment:
how is this socmedArr supposed to look like?

the expected input should be sth like:
[ {name: 'linkedin' , url: 'https://linkedin.com/abc'}, {name: 'github', url: 'https://github.com/abc'}]

your current code won't generate this output right?